src/main/java/com/mongodb/hibernate/jdbc/MongoPreparedStatement.java

```diff
@@ -343,7 +342,7 @@ boolean isUsed() {
      * <p>Note that only find expression is involved before HIBERNATE-74. TODO-HIBERNATE-74 delete this temporary method
      */
     private static void checkComparatorNotComparingWithNullValues(BsonDocument document) {
-        var comparisonOperators = Set.of("$eq", "$ne", "$gt", "$gte", "$lt", "$lte", "$in", "$nin");
+        var comparisonOperators = Set.of("$ne", "$gt", "$gte", "$lt", "$lte", "$in", "$nin");
```

Comment:
I removed `$eq` from here, so that I could use `null`s in `FunctionForArrayIntegrationTests` (for now this seems better than removing the protection completely). As a result of this change, I had to make the change to `SimpleSelectQueryIntegrationTests.java`.